Fuzzing with `multiple` mode not working

### Nuclei version: 3.2.4

### Current Behavior:
Currently nuclei panic when using fuzzing in `multiple` mode.

### Expected Behavior:
Nuclei's multiple mode fuzzing should work without panic.

### Steps To Reproduce:
__test.yaml__
```yaml
id: log4j-header-injection

info:
  name: Log-4j Injection In Headers
  author: Nishantbhagat57
  severity: high
  description: Detects potential Out-of-Band Log-4j Injection vulnerabilities, blind testing using payloads in headers.
  reference:
    - https://www.acunetix.com/blog/web-security-zone/critical-alert-log4shell-cve-2021-44228-in-log4j-possibly-the-biggest-impact-vulnerability-ever/
  tags: cmdi, oast, dast, log4j

http:
  - payloads:
      inject:
        - aa

    fuzzing:
      - part: header
        type: replace
        mode: multiple
        fuzz:
          User-Agent: "{{inject}}"
          Referer: "{{inject}}"
          X-Client-IP: "{{inject}}"
          X-Forwarded-For: "{{inject}}"
          X-Api-Version: "{{inject}}"
          
    matchers-condition: or
    matchers:
      - type: word
        words:
          - "test"
```

```console


✗ nuclei -t test.yaml -u example.com -dast -proxy http://localhost:8080

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v3.2.4

                projectdiscovery.io

[VER] Using http://localhost:8080 as proxy server
[VER] Started metrics server at localhost:9092
[INF] Current nuclei version: v3.2.4 (latest)
[INF] Current nuclei-templates version: v9.8.5 (latest)
[WRN] Scan results upload to cloud is disabled.
[INF] New templates added in latest release: 142
[INF] Templates loaded for current scan: 1
[WRN] Loading 1 unsigned templates for scan. Use with caution.
[INF] Targets loaded for current scan: 1
[INF] Running httpx on input host
[INF] Found 1 URL from httpx
panic: interface conversion: interface {} is map[string]interface {}, not string

goroutine 118 [running]:
github.com/projectdiscovery/nuclei/v3/pkg/fuzz/component.(*Header).Rebuild.func1({0x0, 0x0}, {0x1056a1660, 0x14002f19770})
        github.com/projectdiscovery/nuclei/v3/pkg/fuzz/component/headers.go:99 +0x1b8
github.com/projectdiscovery/nuclei/v3/pkg/fuzz/dataformat.(*KV).Iterate(0x14002f19650?, 0x140047c3388)
        github.com/projectdiscovery/nuclei/v3/pkg/fuzz/dataformat/kv.go:75 +0xbc
github.com/projectdiscovery/nuclei/v3/pkg/fuzz/component.(*Header).Rebuild(0x140057e4470)
        github.com/projectdiscovery/nuclei/v3/pkg/fuzz/component/headers.go:85 +0x6c
github.com/projectdiscovery/nuclei/v3/pkg/fuzz.(*Rule).executeRuleValues(0x140007670e0, 0x14000d0a3c0, {0x105b76b20, 0x140057e4470})
        github.com/projectdiscovery/nuclei/v3/pkg/fuzz/execute.go:214 +0xb8
github.com/projectdiscovery/nuclei/v3/pkg/fuzz.(*Rule).Execute(0x140007670e0, 0x14000d0a3c0)
        github.com/projectdiscovery/nuclei/v3/pkg/fuzz/execute.go:132 +0x9dc
```

--- a/cmd/integration-test/fuzz.go
+++ b/cmd/integration-test/fuzz.go
@@ -18,6 +18,7 @@ const (
 
 var fuzzingTestCases = []TestCaseInfo{
 	{Path: "fuzz/fuzz-mode.yaml", TestCase: &fuzzModeOverride{}},
+	{Path: "fuzz/fuzz-multi-mode.yaml", TestCase: &fuzzMultipleMode{}},
 	{Path: "fuzz/fuzz-type.yaml", TestCase: &fuzzTypeOverride{}},
 	{Path: "fuzz/fuzz-query.yaml", TestCase: &httpFuzzQuery{}},
 	{Path: "fuzz/fuzz-headless.yaml", TestCase: &HeadlessFuzzingQuery{}},
@@ -173,4 +174,30 @@ func (h *HeadlessFuzzingQuery) Execute(filePath string) error {
 		return err
 	}
 	return expectResultsCount(got, 2)
+}
+
+type fuzzMultipleMode struct{}
+
+// Execute executes a test case and returns an error if occurred
+func (h *fuzzMultipleMode) Execute(filePath string) error {
+	router := httprouter.New()
+	router.GET("/", func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+		xClientId := r.Header.Get("X-Client-Id")
+		xSecretId := r.Header.Get("X-Secret-Id")
+		if xClientId != "nuclei-v3" || xSecretId != "nuclei-v3" {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+		w.Header().Set("Content-Type", "text/html")
+		resp := fmt.Sprintf("<html><body><h1>This is multi-mode fuzzing test: %v <h1></body></html>", xClientId)
+		fmt.Fprint(w, resp)
+	})
+	ts := httptest.NewTLSServer(router)
+	defer ts.Close()
+
+	got, err := testutils.RunNucleiTemplateAndGetResults(filePath, ts.URL+"?url=https://scanme.sh", debug, "-jsonl", "-fuzz")
+	if err != nil {
+		return err
+	}
+	return expectResultsCount(got, 1)
 }

--- a/integration_tests/fuzz/fuzz-multi-mode.yaml
+++ b/integration_tests/fuzz/fuzz-multi-mode.yaml
@@ -1,0 +1,27 @@
+id: fuzz-multi-mode-test
+
+info:
+  name: multi-mode fuzzing test
+  author: pdteam
+  severity: info
+
+http:
+  - payloads:
+      inject:
+        - nuclei-v1
+        - nuclei-v2
+        - nuclei-v3
+
+    fuzzing:
+      - part: header
+        type: replace
+        mode: multiple
+        fuzz:
+          X-Client-Id: "{{inject}}"
+          X-Secret-Id: "{{inject}}"
+          
+    matchers-condition: or
+    matchers:
+      - type: word
+        words:
+          - "nuclei-v3"

--- a/pkg/fuzz/component/headers.go
+++ b/pkg/fuzz/component/headers.go
@@ -83,6 +83,9 @@ func (q *Header) Delete(key string) error {
 func (q *Header) Rebuild() (*retryablehttp.Request, error) {
 	cloned := q.req.Clone(context.Background())
 	q.value.parsed.Iterate(func(key string, value any) bool {
+		if strings.TrimSpace(key) == "" {
+			return true
+		}
 		if strings.EqualFold(key, "Host") {
 			return true
 		}

--- a/pkg/fuzz/execute.go
+++ b/pkg/fuzz/execute.go
@@ -211,6 +211,14 @@ func (rule *Rule) executeRuleValues(input *ExecuteRuleInput, ruleComponent compo
 		})
 		// if mode is multiple now build and execute it
 		if rule.modeType == multipleModeType {
+			rule.Fuzz.KV.Iterate(func(key, value string) bool {
+				var evaluated string
+				evaluated, input.InteractURLs = rule.executeEvaluate(input, key, "", value, input.InteractURLs)
+				if err := ruleComponent.SetValue(key, evaluated); err != nil {
+					return true
+				}
+				return true
+			})
 			req, err := ruleComponent.Rebuild()
 			if err != nil {
 				return err

